File: Scripts/CS/PlayerScripting/Modules/NetworkModule.lua
---@class NetworkModule
local NetworkModule = {}


---@param sectorName string
---@return any
function NetworkModule.LinkSector(sectorName) end


---@param sectorName string
---@return any
function NetworkModule.UnlinkSector(sectorName) end


---@param verbose boolean
---@param IP string
---@param MAX_DEPTH number
---@return string[]
function NetworkModule.Scan(verbose, IP, MAX_DEPTH) end


---@param connectedOnly boolean
---@return string[]
function NetworkModule.Sector(connectedOnly) end

return NetworkModule

File: Scripts/CS/PlayerScripting/Modules/NetworkModule.cs
using Godot;

public class NetworkModule {
    public string GetMyIP() {
        return NetworkManager.PlayerNode.IP;
    }
    public CError LinkSector(string sectorName) {
        return NetworkManager.ConnectToSector(sectorName);
    }
    public CError UnlinkSector(string sectorName) {
        return NetworkManager.DisconnectFromSector(sectorName);
    }
    public CError ConnectNode(string IP) {
        return ShellCore.ConnectNode(IP);
    }
    public string[] Scan(string IP = "") {
        return ShellCore.Scan(IP);
    }
    public string[] Sector(bool connectedOnly = false, int level=-1) {
        return NetworkManager.GetSectorNames(connectedOnly, level);
    }
    public int? GetNodeFirewallRating(string IP) {
        return NetworkManager.GetNodeByIP(IP)?.DefLvl;
    }
    public SecLvl? GetNodeSecurity(string IP) {
        return NetworkManager.GetNodeByIP(IP)?.SecLvl;
    }
}